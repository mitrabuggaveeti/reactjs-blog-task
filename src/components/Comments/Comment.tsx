import React from 'react'
import { InewState as Props} from '../Post/Types';

interface Iprops {
    comments : Props['comment']['data']
}    

const Comment : React.FC<Iprops> = (props) => {

    const renderAllComments : JSX.Element[] =  props.comments.map((comment) =>{
        return (
          <li key={comment.id}>
              {comment.body}
              <p> - {comment.email}</p>
         </li>
        )
    });

    return (
        <div>
            <h2>Comments</h2>
            <ul data-testid='allComments'>
                {
                    renderAllComments
                }
            </ul>
        </div>
    )
}

export default Comment
