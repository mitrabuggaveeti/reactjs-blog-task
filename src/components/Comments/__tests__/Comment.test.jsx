import { cleanup, render,waitFor} from "@testing-library/react";
import Comment from "../Comment";
 
export const MOCK_DATA = {
    data :[
     {
         "postId": 1,
         "id": 1,
         "name": "id labore ex et",
         "email": "Eliseo@gardner.biz",
         "body": "laudantium enim quasi est"
       },
       {
         "postId": 1,
         "id": 2,
         "name": "quo vero rei",
         "email": "Jayne_Kuhic@sydney.com",
         "body": "est natus enim nihil"
       }
     ]
 }

describe('Comment component should', () => {
    afterEach(cleanup);

    it('Render',async () => {
                
        const { getByTestId, asFragment } = render(<Comment comments={MOCK_DATA.data}/>);
        const listNode = await waitFor(() => getByTestId('allComments'));

        expect(asFragment()).toMatchSnapshot();
        expect(listNode).toBeTruthy();
    });
})
