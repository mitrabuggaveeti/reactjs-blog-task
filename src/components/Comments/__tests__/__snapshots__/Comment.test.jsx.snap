// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component should Render 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Comments
    </h2>
    <ul
      data-testid="allComments"
    >
      <li>
        laudantium enim quasi est
        <p>
           - Eliseo@gardner.biz
        </p>
      </li>
      <li>
        est natus enim nihil
        <p>
           - Jayne_Kuhic@sydney.com
        </p>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
